sbom/formatted_reader.go

```diff
@@ -78,6 +87,62 @@ func (f *FormattedReader) Read(b []byte) (int, error) {
 			}
 		}
 
+		// Makes SPDX SBOM more reproducible, see
+		// https://github.com/paketo-buildpacks/packit/issues/368 for more details.
+		if f.format.ID() == "spdx-2-json" {
+			hash, err := hashstructure.Hash(f.sbom.syft, hashstructure.FormatV2, nil)
+			if err != nil {
+				// not tested
+				return 0, fmt.Errorf("failed to modify SPDX SBOM for reproducibility: %w", err)
+			}
+			hashBytes := make([]byte, binary.MaxVarintLen64)
+			binary.PutUvarint(hashBytes, hash)
+
+			var spdxOutput map[string]interface{}
+
+			err = json.Unmarshal(output, &spdxOutput)
+			if err != nil {
+				return 0, fmt.Errorf("failed to modify SPDX SBOM for reproducibility: %w", err)
+			}
+			for k := range spdxOutput {
+				if k == "documentNamespace" {
+					uri, err := url.Parse(spdxOutput[k].(string))
+					if err != nil {
+						// not tested
+						return 0, err
+					}
+
+					uri.Host = "paketo.io"
+					uri.Path = strings.Replace(uri.Path, "syft", "packit", 1)
+					oldBase := filepath.Base(uri.Path)
+					source, _, _ := strings.Cut(oldBase, "-")
+					newBase := fmt.Sprintf("%s-%s", source, uuid.NewSHA1(uuid.NameSpaceURL, hashBytes))
+					uri.Path = strings.Replace(uri.Path, oldBase, newBase, 1)
+
+					spdxOutput[k] = uri.String()
+				}
+				if k == "creationInfo" {
+					creationInfo := spdxOutput["creationInfo"].(map[string]interface{})
+                                         creationInfo["created"] = time.Time{} // This is the zero-valued time
+
+					source_date_epoch := os.Getenv("SOURCE_DATE_EPOCH")
+					if source_date_epoch != "" {
+						sde, err := strconv.ParseInt(source_date_epoch, 10, 64)
+						if err != nil {
+							// not tested
+							return 0, fmt.Errorf("failed to parse SOURCE_DATE_EPOCH: %w", err)
+						}
+						creationInfo["created"] = time.Unix(sde, 0)
```

Comment:
```suggestion
						creationInfo["created"] = time.Unix(sde, 0).UTC()
```
Resolves https://github.com/paketo-buildpacks/packit/pull/372#discussion_r937045461